packages/stats/gbstats/gbstats.py

```diff
@@ -385,3 +421,117 @@ def check_srm(users, weights):
+    var_names = analysis.var_names
+    weights = analysis.weights
+    max_dimensions = analysis.max_dimensions
+    baseline_index = analysis.baseline_index
+    stats_engine = (
+        StatsEngine.FREQUENTIST
+        if analysis.stats_engine == "frequentist"
+        else StatsEngine.BAYESIAN
+    )
+
+    # If we're doing a daily time series, we need to diff the data
+    if analysis.dimension == "pre:datedaily":
+        rows = diff_for_daily_time_series(rows)
+
+    # Convert raw SQL result into a dataframe of dimensions
+    df = get_metric_df(
+        rows=rows,
+        var_id_map=var_id_map,
+        var_names=var_names,
+    )
+
+    # Limit to the top X dimensions with the most users
+    reduced = reduce_dimensionality(
+        df=df,
+        max=max_dimensions,
+    )
+
+    # Get the stats engine configuration for this analysis
+    engine_config: Dict[str, Union[int, float, DifferenceType]] = {
+        "phase_length_days": analysis.phase_length_days
+    }
+
+    if analysis.difference_type == "absolute":
+        engine_config["difference_type"] = DifferenceType.ABSOLUTE
+    elif analysis.difference_type == "scaled":
+        engine_config["difference_type"] = DifferenceType.SCALED
+    else:
+        engine_config["difference_type"] = DifferenceType.RELATIVE
+
+    if stats_engine == StatsEngine.FREQUENTIST and analysis.sequential_testing_enabled:
+        engine_config["sequential"] = True
+        engine_config[
+            "sequential_tuning_parameter"
+        ] = analysis.sequential_tuning_parameter
+
+    if stats_engine == StatsEngine.FREQUENTIST and analysis.alpha:
+        engine_config["alpha"] = analysis.alpha
+
+    # Run the analysis for each variation and dimension
+    result = analyze_metric_df(
+        df=reduced,
+        weights=weights,
+        inverse=inverse,
+        engine=stats_engine,
+        engine_config=engine_config,
+    )
+
+    return format_results(result, baseline_index)
+
+
+def process_single_metric(
+    mdata: MetricDataForStatsEngine,
+    analyses: List[AnalysisSettingsForStatsEngine],
+    var_id_map: Dict[str, int],
+):
+    rows = pd.DataFrame(mdata.rows)
+    inverse = mdata.inverse
+    multiple_exposures = mdata.multiple_exposures
+
+    # Detect any variations that are not in the returned metric rows
+    unknown_var_ids = detect_unknown_variations(rows=rows, var_id_map=var_id_map)
+
+    results = [
+        process_analysis(
+            rows=rows,
+            inverse=inverse,
+            var_id_map=var_id_map,
+            analysis=a,
+        )
+        for a in analyses
+    ]
+
+    return {
+        "metric": mdata.metric,
+        "analyses": [
+            {
+                "unknownVariations": list(unknown_var_ids),
+                "dimensions": r,
+                "multipleExposures": multiple_exposures,
+            }
+            for r in results
+        ],
+    }
+
+
+def process_experiment_results(data: Dict[str, Any]):
+    d = DataForStatsEngine(
```

Comment:
We could also do this in `stats.ts`, but I'm not sure which is better for error handling.